### PR TITLE
Choose correct address family for UDP socket when `udp-bind` is not specified explicitly

### PR DESCRIPTION
**Pull request by dlon**
`udp-bind` defaults to `0.0.0.0` for `tcp2udp`. This means that `AF_INET` is used as the address family even if the destination is an IPv6 address. This PR fixes this by binding the socket to `::` by default if the destination is an IPv6 address.

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/udp-over-tcp/28)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -365,9 +365,9 @@ dependencies = [
 
 [[package]]
 name = "regex-syntax"
-version = "0.6.25"
+version = "0.6.27"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "f497285884f3fcff424ffc933e56d7cbca511def0c9831a7f9b5f6153e3cc89b"
+checksum = "a3f87b73ce11b1619a3c6332f45341e0047173771e8b8b73f87bfeefb7b56244"
```
Comment (faern) [inline Cargo.lock:370]: Are these upgrades of `regex` and `regex-syntax` relevant to the UDP bind fix? Otherwise I think we should remove them from this PR. If we need to upgrade dependencies we can do so in a separate PR. But I see no reason to do that unless we want to build a release, because the lockfile is not used when consumed as a library.

**Review by faern — CHANGES_REQUESTED** [review]
Looks good overall. I assume you have checked so both the CLI stays sane with these changes and that it's ergonomic to consume as a library.

**Review by faern — COMMENTED** [review]
Please mention this in the changelog! It's a simplification on how the tcp2udp server can be used

> *[Reviewable](https://reviewable.io/reviews/mullvad/udp-over-tcp/28#-:-N7UkxRm5YcUmwgRCCwd:b72wrjg)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @dlon and @faern)

**Review by dlon — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/udp-over-tcp/28)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @faern)

___
*[`Cargo.lock` line 370 at r1](https://reviewable.io/reviews/mullvad/udp-over-tcp/28#-N6hPeYIGUQ6F7SeQ1zT-r1-370:-N7UrVheBZF9g6U_bNRA:b-d4p0pn) ([raw file](https://github.com/mullvad/udp-over-tcp/blob/7b14eb348fb454dcae92f7dc015b220b0c629de4/Cargo.lock#L370)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Are these upgrades of `regex` and `regex-syntax` relevant to the UDP bind fix? Otherwise I think we should remove them from this PR. If we need to upgrade dependencies we can do so in a separate PR. But I see no reason to do that unless we want to build a release of the server binary, because the lockfile is not used when consumed as a library.
</blockquote></details>

Fixed in another PR.

**Review by faern — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/udp-over-tcp/28#-:-N7VAYgN-QZ0lM_e9kBf:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dlon)
